src/app/query-explainer/query-explainer.component.ts

```diff
@@ -34,26 +43,31 @@ export class QueryExplainerComponent extends BaseComponent implements OnInit {
 
     /* istanbul ignore next */
     ngOnInit(): void {
-        const commitmentUploadToken = this.extractCommitmentUploadToken();
-        if (commitmentUploadToken) {
-            this.componentData$ = this.commitmentDataWithoutOutput(commitmentUploadToken).pipe(
-                switchMap((response: QueryExplainerResponse) => {
-                    const cloneData = Object.assign({}, response);
-                    if ("output" in cloneData) {
-                        delete cloneData.output;
-                    }
-                    return combineLatest([of(response), this.queryExplainerService.verifyQuery(cloneData)]).pipe(
-                        map(([sqlQueryExplainerResponse, sqlQueryVerify]) => ({
-                            sqlQueryExplainerResponse,
-                            sqlQueryVerify,
-                        })),
+        this.commitmentUploadToken = this.extractCommitmentUploadToken();
+        if (this.commitmentUploadToken) {
+            this.componentData$ = this.uploadCommitment(this.commitmentUploadToken).pipe(
+                switchMap((uploadedCommitment: QueryExplainerResponse) => {
+                    return combineLatest([
+                        of(uploadedCommitment),
+                        this.queryExplainerService.processQueryWithSchema(uploadedCommitment.input.query),
+                        this.queryExplainerService.verifyQuery(uploadedCommitment),
+                    ]).pipe(
+                        map(([commitment, dataJsonAoS, sqlQueryVerify]) => {
+                            return {
+                                sqlQueryExplainerResponse: {
+                                    ...commitment,
+                                    ...dataJsonAoS,
+                                },
+                                sqlQueryVerify,
+                            };
+                        }),
                     );
                 }),
             );
         }
     }
 
-    private commitmentDataWithoutOutput(commitmentUploadToken: string): Observable<QueryExplainerResponse> {
+    private uploadCommitment(commitmentUploadToken: string): Observable<QueryExplainerResponse> {
```

Comment:
Seems like it should be "download", not "upload".

Reply:
Cahnged